src/main/java/bio/terra/cli/command/helperclasses/CommandSetup.java

```diff
@@ -0,0 +1,49 @@
+package bio.terra.cli.command.helperclasses;
+
+import bio.terra.cli.auth.AuthenticationManager;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import java.io.PrintStream;
+import java.util.concurrent.Callable;
+
+public abstract class CommandSetup implements Callable<Integer> {
+  protected GlobalContext globalContext;
+  protected WorkspaceContext workspaceContext;
+
+  // output stream to use for writing command return value
+  protected static final PrintStream OUT = System.out;
+
+  @Override
+  public Integer call() {
+    // read in the global and workspace context
+    globalContext = GlobalContext.readFromFile();
+    workspaceContext = WorkspaceContext.readFromFile();
+
+    // do the login flow if required
+    if (doLogin()) {
+      new AuthenticationManager(globalContext, workspaceContext).loginTerraUser();
+    }
+
+    // execute the command
+    execute();
+
+    // set the command exit code
+    return 0;
+  }
+
```

Comment:
A nice side effect of this change is that the `void` return on `execute()` makes it unambiguous that commands should be using exceptions instead of return codes for errors (as a newbie to CLI repo I had to go look at Main to make sure).  It might be worth adding a comment to the effect that `main()` will handle exceptions thrown by commands to the docs here.

Reply:
Good idea. Added comment.